internal/os/os.go

```diff
@@ -41,9 +38,10 @@ type OS struct {
 	LastUpgradeTime                 string
 	GracefulRebootChannel           chan struct{}
 	GracefulRebootCompletionChannel chan struct{}
+	OsExecCommands                  OsExecCommands
 }
 
-func NewOS(gracefulRebootChannel chan struct{}) *OS {
+func NewOS(gracefulRebootChannel chan struct{}, osExecCommands OsExecCommands) *OS {
```

Comment:
As mentioned earlier, we can skip the requirement for a second argument and initialize OsExecCommands in this method.